File: Principal/DataBase/Dominio/SolicitacaoDominio.cs
using DigoFramework.Json;
using System.Collections.Generic;

namespace RealLife.DataBase.Dominio
{
    internal class SolicitacaoDominio : DominioRealLifeBase
    {
        #region Constantes

        public enum EnmMetodo
        {
            APARENCIA_SALVAR,

            CONTA_SALVAR,

            DESCONHECIDO,

            LOGIN_ENTRAR,
        }

        #endregion Constantes

        #region Atributos

        private object[] _arrObjArgumento;
        private EnmMetodo _enmMetodo = EnmMetodo.DESCONHECIDO;

        public object[] arrObjArgumento
        {
            get
            {
                return _arrObjArgumento;
            }

            set
            {
                _arrObjArgumento = value;
            }
        }

        public EnmMetodo enmMetodo
        {
            get
            {
                return _enmMetodo;
            }

            set
            {
                _enmMetodo = value;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        public T getObjArgumento<T>(int intIndex = 0)
        {
            if (intIndex < 0)
            {
                return default(T);
            }

            if (this.arrObjArgumento == null)
            {
                return default(T);
            }

            if (this.arrObjArgumento.Length < (intIndex + 1))
            {
                return default(T);
            }

            var jsn = this.arrObjArgumento[intIndex].ToString();

            if (string.IsNullOrEmpty(jsn))
            {
                return default(T);
            }

            return Json.i.fromJson<T>(jsn);
        }

        internal void addArgumento(params object[] arrObjArgumento)
        {
            if (arrObjArgumento == null)
            {
                return;
            }

            if (arrObjArgumento.Length < 1)
            {
                return;
            }

            var lstObjArgumento = new List<object>(this.arrObjArgumento);

            foreach (var objArgumento in arrObjArgumento)
            {
                lstObjArgumento.Add(objArgumento);
            }

            this.arrObjArgumento = lstObjArgumento.ToArray();
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: RealLifeShared.TypeScript/dominio/SolicitacaoDominio.ts
/// <reference path="../enumerado/EnmMetodo.ts"/>
/// <reference path="DominioRealLifeBase.ts"/>

module RealLifeShared
{
    // #region Importações
    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class SolicitacaoDominio extends DominioRealLifeBase
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _arrObjArgumento: Array<Object>;
        private _enmMetodo: EnmMetodo = EnmMetodo.DESCONHECIDO;
        private _fncRetorno: Function;

        public get arrObjArgumento(): Array<Object>
        {
            return this._arrObjArgumento;
        }

        public set arrObjArgumento(arrObjArgumento: Array<Object>)
        {
            this._arrObjArgumento = arrObjArgumento;
        }

        public get enmMetodo(): EnmMetodo
        {
            return this._enmMetodo;
        }

        public set enmMetodo(enmMetodo: EnmMetodo)
        {
            this._enmMetodo = enmMetodo;
        }

        public get fncRetorno(): Function
        {
            return this._fncRetorno;
        }

        public set fncRetorno(fncRetorno: Function)
        {
            this._fncRetorno = fncRetorno;
        }

        // #endregion Atributos

        // #region Construtores

        constructor(enmMetodo: EnmMetodo, fncRetorno: Function = null, ...arrObjArgumento: Object[])
        {
            super();

            this.enmMetodo = enmMetodo;
            this.fncRetorno = fncRetorno;

            if (arrObjArgumento != null)
            {
                arrObjArgumento.forEach((objArgumento) => { this.addObjArgumento(objArgumento); });
            }
        }

        // #endregion Construtores

        // #region Métodos

        public addObjArgumento(objArgumento: Object): void
        {
            if (objArgumento == null)
            {
                return;
            }

            if (this.arrObjArgumento == null)
            {
                this.arrObjArgumento = new Array<Object>();
            }

            this.arrObjArgumento.push(JSON.stringify(objArgumento));
        }

        public processarResposta(objResposta: RespostaDominio): boolean
        {
            if (objResposta.intSolicitacaoId != this.intId)
            {
                return false;
            }

            if (this.fncRetorno == null)
            {
                return true;
            }

            this.fncRetorno(objResposta);

            return true;
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}